### ISSUE TITLE
Unknown parameter: max_bin_by_feature

### ISSUE THREAD

**Issue by trashprogrammer** [microsoft/LightGBM#2425]
Hello, I'm getting an error that it doesn't seem to recognize the max_bin_by_feature. 
If anyone can help me out, that'd be great. Thank you

###Here is the code:

lgb_train = lgb.Dataset(train.iloc[:,0:60], train['y']) 
params = {
    'boosting_type': 'dart',
    'objective': 'regression',
     'metric': {None},
    'num_leaves': 45,
    'learning_rate': 0.5,
    'drop_rate': 0.032,
    'uniform_drop': False,
     'min_data_in_leaf':100,
    'feature_fraction': 0.9,
    'bagging_fraction': 0.1,
     'lambda_l2':0.05,
     'max_depth':12,
    'verbose': 0,
    'max_bin_by_feature':(list(np.ones(30)*100)+list(np.ones(30)*240)),
    'num_threads' : 36,
    'seed':199
}
gbm2 = lgb.train(params,
                lgb_train,
                num_boost_round=5)

LightGBM version:
lightgbm-2.2.3 

## Error message
[LightGBM] [Warning] Unknown parameter: max_bin_by_feature

**Comment by StrikerRUS** [comment]
@trashprogrammer Hi!

`max_bin_by_feature` was introduced after `2.2.3` release. You can build LightGBM from the latest sources or wait a little bit for #2138.
